diff --git a/grammar.js b/grammar.js
index 7ca23ff..8221a3e 100644
--- a/grammar.js
+++ b/grammar.js
@@ -749,6 +749,7 @@ module.exports = grammar({
         repeat(choice(
           alias($.unescaped_double_jsx_string_fragment, $.string_fragment),
           $.html_character_reference,
+          alias(token.immediate('&'), $.string_fragment),
         )),
         '"',
       ),
@@ -757,6 +758,7 @@ module.exports = grammar({
         repeat(choice(
           alias($.unescaped_single_jsx_string_fragment, $.string_fragment),
           $.html_character_reference,
+          alias(token.immediate('&'), $.string_fragment),
         )),
         '\'',
       ),
@@ -766,10 +768,10 @@ module.exports = grammar({
     // We give names to the token() constructs containing a regexp
     // so as to obtain a node in the CST.
     //
-    unescaped_double_jsx_string_fragment: _ => token.immediate(prec(1, /([^"&]|&[^#A-Za-z])+/)),
+    unescaped_double_jsx_string_fragment: _ => token.immediate(prec(1, /([^"&]|&[^#"A-Za-z])+/)),
 
     // same here
-    unescaped_single_jsx_string_fragment: _ => token.immediate(prec(1, /([^'&]|&[^#A-Za-z])+/)),
+    unescaped_single_jsx_string_fragment: _ => token.immediate(prec(1, /([^'&]|&[^#'A-Za-z])+/)),
 
     _jsx_attribute_value: $ => choice(
       alias($._jsx_string, $.string),
